### DISCUSSION TITLE
Integrate Ollama tool calling

### DISCUSSION THREAD

**Discussion by isr431** [open-webui/open-webui#4179] [Ideas]
I'm still a beginner to this stuff, so please correct me if I'm wrong. Ollama v0.3 now supports tool calling. Is it possible for OpenWebUI to have an interface for adding and using tools? Users can add their own functions in Python/JavaScript, and have to option to enable/disable them for each chat.
I'm not sure how different this is to the 'Pipelines' feature that OpenWebUI has.

**Comment by justinh-rahb** [comment]
We already have a [Tools](https://openwebui.com/tools) and [Functions](https://openwebui.com/functions) feature that predates this addition to Ollama's API, and does not rely on it. In fact it's basically API-agnostic and will work with any model that is trained for functions/tool-calling, in theory.

**Reply by caymaynard** [reply]
Hi @justinh-rahb , has this been integrated in the HTTP API?
